java/client/src/main/java/glide/api/commands/ListBaseCommands.java

```diff
@@ -369,4 +373,44 @@ CompletableFuture<Long> linsert(
      * }</pre>
      */
     CompletableFuture<Long> lpushx(String key, String[] elements);
+
+    /**
+     * Pops one or more elements from the first non-empty list from the list of provided <code>key
+     * </code>.
+     *
+     * @since Redis 7.0 and above.
+     * @apiNote When in cluster mode, all <code>keys</code> must map to the same hash slot.
+     * @see <a href="https://valkey.io/commands/lmpop/">valkey.io</a> for details.
+     * @param keys The list of provided <code>key</code> names.
+     * @param direction The direction based on which elements are popped from - see {@link
+     *     PopDirection}.
+     * @param count The count of elements to be popped and returned.
+     * @return A <code>Map</code> of <code>key</code> name mapped array of popped elements.
+     * @example
+     *     <pre>{@code
+     * client.lpush("testKey", new String[] {"one", "two", "three"}.;
+     * String[] result = client.lmpop(new String[] {"testKey"}, PopDirection.LEFT, 1L).get().get("testKey");
```

Comment:
Recommend doing this, to show that that the return type is a Map. 
It's also weird to have the `.get().get()` which might be confusing to some people. 
```suggestion
     * Map<String, String[]> result = client.lmpop(new String[] {"testKey"}, PopDirection.LEFT, 1L).get();
     * String resultValue = result.get("testKey");
```